src/firmware/app/control/trajectory_planner.h

```diff
@@ -181,123 +213,177 @@ static void app_trajectory_planner_generateConstArclengthTrajectoryPositions(
+ * point on the path can the sum of centripetal and acceleration force be greater than
+ * 'max_allowable_linear_acceleration'
  *
- * @pre The input velocity_profile is required to be FORWARDS CONTINUOUS before it is used
- * as an input to this function
- * @pre forwards_continuous_velocity_profile is pre-allocated up to at least
+ *  PositionTrajectory generation is done by assuming constant acceleration capability.
+ *      - The generator will assume maximum acceleration for the robot between each
+ * segment on the path. If using max acceleration breaches either the speed limit of the
+ * robot, or the centripetal acceleration limit defined by the curvature and robot speed -
+ * the planner will assume the max acceleration that will remain bellow this limit.
+ *
+ *     - To ensure the robot is capable of deccelerating to react to any sudden changes in
+ * the path, the trajectory is checked for backwards continuity
+ *
+ * NOTE: The following function is based on an algorithm described in page 18 of
+ * http://www2.informatik.uni-freiburg.de/~lau/students/Sprunk2008.pdf
+ *
+ * @pre path_parameters.num_segments > 2 and num_segments <=
  * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * @pre path_parameters.max_allowable_linear_acceleration > 0
+ * @pre path_parameters.max_allowable_linear_speed > 0
+ * @pre path_parameters.init_speed >= 0
+ * @pre path_parameters.final_linear_speed >= 0
+ * @pre The trajectory.trajectory_elements[] array is pre-allocated to handle up to the
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS limit
+ *
+ * @param trajectory [in/out] The trajectory with pre-initialized path parameters. Will be
+modified in-place
  *
- * @param num_segments [in] The number of segments(elements) in the velocity_profile array
- * @param forwards_continuous_velocity_profile [in/out] The forwards continuous velocity
- * profile to ebe modified in-place into a profile that is also backwards continuous
- * @param arc_segment_length [in] The arc segment length of each segment in the path the
- * profile is being generated for in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point on
- * the velocity profile
+ * @return The generation status. Can be OK or error message.
  */
-void app_trajectory_planner_generateBackwardsContinuousVelocityProfile(
-    const unsigned int num_segments,
-    float forwards_continuous_velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const float arc_segment_length, const float max_allowable_acceleration);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_generateConstantParameterizationPositionTrajectory(
+    PositionTrajectory_t* trajectory);
 
 /**
- *  Adds a time profile to an existing position profile for the given input parameters
+ * Function generates a constant interpolation period trajectory from the path parameters
+ * specified in the constant parameterization trajectory. The function creates a constant
```

Comment:
Could we add a quick doc to the start of this header explaining `constant interpolation`, `constant parametrization`, `constant time`, and `constant arc length`? I think that understanding (roughly) what each of these mean is pretty essential for even being able to understand the interfaces of these functions.

Reply:
Added comment explaining the 3 👍

Reply:
Sorry, meant at the top of the header, since I think it is probably relevant to more then just this function?

Reply:
Moved to the top of the header

Reply:
I don't see it?

Reply:
Hmm, check again 👀